esp/services/ws_machine/ws_machineService.cpp

```diff
@@ -1183,7 +1183,7 @@ int Cws_machineEx::runCommand(IEspContext& context, const char* sAddress, const
         IFRunSSH * connection = createFRunSSH();
         connection->init(command.str(),NULL,NULL,NULL,m_SSHConnectTimeoutSeconds,0);
         // executed as single connection
-        connection->exec(sAddress,NULL,true);
+        connection->exec(sAddress, NULL, false);
```

Comment:
What will be the impact of changing the "background" parameter to false? Does it mean multiple opens do not happen in parallel, for example? That could be an issue on a large cluster

Reply:
If a user selects HPCC components from multiple IPs, the ws_machine code will create one thread for each IP and call this runCommand() (and this exec()) in each thread. So, the multiple exec()'s will be run in parallel.

Reply:
So why did it say true before?

Reply:
This is my guessing: the original code (https://github.com/hpcc-systems/HPCC-Platform/commit/993519e6030832340948813816945401fedfc2a5) does not realize that there is already multi-thread implementation.

Reply:
I think invokeProgram() would run in the foreground ??

Reply:
I think? the exec() code runs the command and waits for it to finish and collects the output, so not really understanding why it was originally set to background when exit code and output of command is wanted.
@wangkx could we test with >1 IP and a long running (ie adding sleep 30) preflight to make sure it does not take longer than with 1 IP ?

Reply:
@richardkchapman fyi